core/src/test/java/org/pentaho/di/connections/ConnectionManagerTest.java

```diff
@@ -61,13 +72,56 @@ public class ConnectionManagerTest {
 
   private ConnectionManager connectionManager;
 
-  private MemoryMetaStore memoryMetaStore = new MemoryMetaStore();
+  private Bowl bowl;
+
+  private MemoryMetaStore memoryMetaStore;
+
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+    KettleClientEnvironment.init();
+    DefaultBowl defaultBowl = DefaultBowl.getInstance();
+    assertNotNull( defaultBowl );
+
+    // Make sure that the default bowl has a meta-store, as this is a requirement of the connection manager,
+    // and otherwise would affect the default connection manager.
+    IMetaStore defaultMemoryMetaStore = new MemoryMetaStore();
+    defaultBowl.setMetastoreSupplier( () -> defaultMemoryMetaStore );
+  }
 
   @Before
   public void setup() throws Exception {
-    KettleClientEnvironment.init();
-    connectionManager = new ConnectionManager();
-    connectionManager.setMetastoreSupplier( () -> memoryMetaStore );
+    bowl = mock( Bowl.class );
+    memoryMetaStore = new MemoryMetaStore();
+    connectionManager = new ConnectionManager( () -> memoryMetaStore, bowl );
+  }
+
+  @Test
+  public void testGetInstanceHasDefaultBowlAndMetaStore() {
+    DefaultBowl defaultBowl = DefaultBowl.getInstance();
+    assertNotNull( defaultBowl );
+
+    ConnectionManager defaultConnectionManager = ConnectionManager.getInstance();
+    assertNotNull( defaultConnectionManager );
+
+    assertSame( defaultBowl, defaultConnectionManager.getBowl() );
```

Comment:
This assertion will fail if it happens in the same JVM as any other test that uses DefaultBowl.setMetastoreSupplier() beforehand. Granted, nothing seems to use that at the moment.

Reply:
Actually, this test class is calling that method. I don't see why it would fail. These classes do not use `DefaultBowl#getConnectionManager()` if that's what you're hinting at.

Reply:
oh good point. Yeah, this will be fine.